lib/rubocop/cop/lint/useless_ruby2_keywords.rb

```diff
@@ -98,14 +98,22 @@ def inspect_sym(node, sym_node)
           return unless node.parent
 
           method_name = sym_node.value
-          definition = node.parent.each_child_node.detect { |n| method_definition(n, method_name) }
+          definition = find_method_definition(node, method_name)
```

Comment:
Previously we only checked node's parent which is `begin` and not `def`, `block` or `numblock` in case of:

```ruby
def foo
end
ruby2_keywords :foo if respond_to?(:ruby2_keywords)
```

```ruby
s(:begin,
  s(:def, :foo,
    s(:args), nil),
  s(:if,
    s(:send, nil, :respond_to?,
      s(:sym, :ruby2_keywords)),
    s(:send, nil, :ruby2_keywords
      s(:sym, :foo)), nil))
```